dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/Filter/BasicQueryTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Linq.Expressions;
+using Microsoft.Extensions.VectorData;
+
+namespace VectorDataSpecificationTests.Filter;
+
+public abstract class BasicQueryTests<TKey>(BasicQueryTests<TKey>.QueryFixture fixture)
+    : BasicFilterTests<TKey>(fixture) where TKey : notnull
+{
+    // Not all of the connectors allow to sort by the Key, so we sort by the Int.
+    protected override List<FilterRecord> GetOrderedRecords(IQueryable<FilterRecord> filtered)
+        => filtered.OrderBy(r => r.Int).ThenByDescending(r => r.String).ToList();
+
+    protected override async Task<List<FilterRecord>> GetResults(IVectorStoreRecordCollection<TKey, FilterRecord> collection,
+        Expression<Func<FilterRecord, bool>> filter, int top)
+    {
+        FilterOptions<FilterRecord> options = new();
+
+        options.Sort
+            .Ascending(r => r.Int)
+            .Descending(r => r.String);
+
+        return await collection.GetAsync(filter, top, options).ToListAsync();
+    }
+
+    [Obsolete("Not used by derived types")]
+    public sealed override Task Legacy_And() => Task.CompletedTask;
+
+    [Obsolete("Not used by derived types")]
+    public sealed override Task Legacy_equality() => Task.CompletedTask;
+
+    [Obsolete("Not used by derived types")]
+    public sealed override Task Legacy_AnyTagEqualTo_array() => Task.CompletedTask;
+
+    [Obsolete("Not used by derived types")]
+    public sealed override Task Legacy_AnyTagEqualTo_List() => Task.CompletedTask;
+
+    public abstract class QueryFixture : BasicFilterTests<TKey>.Fixture
+    {
+        private static readonly Random s_random = new();
+
+        protected override string CollectionName => "QueryTests";
+
+        /// <summary>
+        /// In contrary to the filter tests, the query uses random vectors,
```

Comment:
Why would this be specific to BasicQueryTests, as opposed to the super-class BasicFilterTests? In both cases, the vector is completely irrelevant and the filter is the only important thing...

In any case, as in the other comment, I propose to let this class (and its super-class) really concentrate only on various expression possibilities inside the filter. If we want to have a test that the vectors are irrelevant, we can do that in another test suite which exercises that (and OrderBy, and Top/Skip...).